src/compiler/checker.ts

```diff
@@ -30861,7 +30867,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const jsxOpeningLikeNode = node ;
             const sig = getResolvedSignature(jsxOpeningLikeNode);
             checkDeprecatedSignature(sig, node);
-            checkJsxReturnAssignableToAppropriateBound(getJsxReferenceKind(jsxOpeningLikeNode), getReturnTypeOfSignature(sig), jsxOpeningLikeNode);
+
+            const elementTypeConstraint = getJsxElementTypeTypeAt(jsxOpeningLikeNode);
+            if (elementTypeConstraint !== undefined) {
+                const tagName = jsxOpeningLikeNode.tagName;
+                const tagType = isJsxIntrinsicIdentifier(tagName)
+                    ? getStringLiteralType(tagName.escapedText as string)

```

Comment:
```suggestion
                    ? unescapeLeadingUnderscores(tagName.escapedText)
```

Reply:
FYI `isIntrinsicJsxName` is also not using `unescapeLeadingUnderscores` which is why I didn't use that originally. I don't know the difference so I just trust you that unescaping is the correct thing to do here.

I assume you meant
```suggestion
                    ? getStringLiteralType(unescapeLeadingUnderscores(tagName.escapedText))

```

though?